cloudvolume/cloudvolume.py

```diff
@@ -143,11 +136,22 @@ def create_new_info(cls, num_channels, layer_type, data_type, encoding, resoluti
   @classmethod
   def extract_path(cls, cloudpath):
     """cloudpath: e.g. gs://neuroglancer/DATASET/LAYER/info or s3://..."""
-    match = re.match(r'^(gs|file|s3|boss)://(/?[\d\w_\.\-]+)/([\d\w_\.\-]+)/([\d\w_\.\-]+)/?', cloudpath)
-    return ExtractedPath(*match.groups())
+    protocol_re = r'^(gs|file|s3|boss)://'
```

Comment:
Why so many re.match and re.sub?
You can just do it all in one line
https://github.com/seung-lab/neuroglancer/blob/it_queue/python/neuroglancer/pipeline/storage.py#L63

Reply:
I found that line hard to understand and maybe error prone. I figured processing the prefix, the suffix, and then the part in the middle I was more likely to get right.